app/uk/gov/hmrc/agentsubscriptionfrontend/views/business_name.scala.html

```diff
@@ -19,20 +19,30 @@
 @import uk.gov.hmrc.agentsubscriptionfrontend.models.BusinessName
 @import uk.gov.hmrc.agentsubscriptionfrontend.controllers.routes
 
-@(businessNameForm: Form[BusinessName])(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
+@(businessNameForm: Form[BusinessName], desErrorExists: Boolean = false)(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
 
 @uk.gov.hmrc.agentsubscriptionfrontend.views.html.main_template(appConfig, title =  error_prefix(businessNameForm) +  Messages("businessName.title"), bodyClasses = None) {
-    <a href="@routes.SubscriptionController.showCheckAnswers()" class="link-back">@Messages("button.back")</a>
+    @if(desErrorExists) {
+        <a href="@routes.BusinessIdentificationController.showConfirmBusinessForm()" class="link-back">@Messages("button.back")</a>
+    } else {
+        <a href="@routes.SubscriptionController.showCheckAnswers()" class="link-back">@Messages("button.back")</a>
+    }
 
     @if(businessNameForm.hasErrors) {
         @errorSummary(Messages("error.summary.heading"), businessNameForm, Seq.empty, Some("business-name"))
     }
-    <h1 class="heading-xlarge">@Messages("businessName.title")</h1>
+  @if(desErrorExists) {
+      <h1 class="heading-xlarge">@Messages("businessName.updated.title")</h1>
+      <p>@Messages("businessName.updated.p1")</p>
+
+  } else {
+      <h1 class="heading-xlarge">@Messages("businessName.title")</h1>
+  }
 
     <p class="panel panel-border-wide">@Messages("businessName.description")</p>
 
     @form(
-        action = routes.SubscriptionController.submitBusinessNameForm(),
+        action = routes.BusinessIdentificationController.submitBusinessNameForm(),
```

Comment:
What happens if I'm on the `desErrorExists = true` version of this page and submit another invalid business name? Does it get routed back to this page with `desErrorExists = true` or does `desErrorExists` get flipped to false?
Also, does it work vice versa, when I do the same on the `desErrorExists = false` version of the page?